src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int RESIZE_VALUE = 2;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
+        Node<K, V> entry = new Node<>(key, value, null);
+        Node<K, V> existing = table[getHash(key)];
```

Comment:
let's save the result of getHash(key) to the separate variable and use it in the following code
```suggestion
        int index = getHash(key)
        Node<K, V> existing = table[index];
```